### ISSUE TITLE
running production fails

### ISSUE THREAD

**Issue by stocksp** [iaincollins/nextjs-starter#18]
Dev mode works great.
I can push the app to Now.
running production locally crashes.

npm run build ... then

nextjs-starter git:(master) ✗ npm run start

> nextjs-starter@2.9.0 start /Users/paul/projects/nextjs-starter
> node index.js

events.js:161
      throw er; // Unhandled 'error' event
      ^

Error: listen EACCES 0.0.0.0:80
    at Object.exports._errnoException (util.js:1023:11)
    at exports._exceptionWithHostPort (util.js:1046:20)
    at Server._listen2 (net.js:1248:19)
    at listen (net.js:1297:10)
    at Server.listen (net.js:1375:9)
    at Function.listen (/Users/paul/projects/nextjs-starter/node_modules/express/lib/application.js:617:24)
    at app.prepare.then.then.db (/Users/paul/projects/nextjs-starter/index.js:116:10)

npm ERR! Darwin 16.4.0
npm ERR! argv "/Users/paul/.nvm/versions/node/v7.5.0/bin/node" "/Users/paul/.nvm/versions/node/v7.5.0/bin/npm" "run" "start"
npm ERR! node v7.5.0
npm ERR! npm  v4.1.2
npm ERR! code ELIFECYCLE
npm ERR! nextjs-starter@2.9.0 start: `node index.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the nextjs-starter@2.9.0 start script 'node index.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the nextjs-starter package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node index.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs nextjs-starter
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls nextjs-starter
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/paul/projects/nextjs-starter/npm-debug.log
➜  nextjs-starter git:(master) ✗

**Comment by iaincollins** [comment]
Looks like you've got something already running on port 80 locally 🙂

> Error: listen EACCES 0.0.0.0:80

This is a not very good error that is thrown (that the app spits out), but that's what it's trying to say anyway.

> npm ERR! Darwin 16.4.0

From this am assuming you are on a Mac, and it's running Apache locally. It runs Apache and a few other services locally so those ports aren't free.

The good news is you can still easily run in production mode locally but on a different port by specifying a PORT override as an ENV VAR:

> npm run build
> PORT=3000 npm start

Notes:

* Each time you change something in build you will want to re-run `npm run build`.
* If you change the CSS and want it to show up in production you should bump the build number in Package.json too (e.g. 1.0.0 to 1.0.1) as it's currently used in CSS versioning.

**Comment by iaincollins** [comment]
Have updated the docs to make this clearer along with a few other improvements. Hope this helps!

**Comment by stocksp** [comment]
Thank you so much for your help AND this template.
Very strange about the port. I'm not running Apache ... I did have a couple of other node apps running on different ports (and Mongodb runnning its strange port).   Still get that failure . But setting the PORT env before running allows it to run.  Yahoo!
I guess its about time to reboot my mac and see if that free's up port 80 ...

Thanks again for your boiler plate!!  Its sooo much more valuable and instructive than the simple examples in the next repo.   I only found it from reading one of the issues ... it really should be in the main readme..
